tap_amazon_ads/discover.py

```diff
@@ -0,0 +1,39 @@
+import singer
+from singer import metadata
+from singer.catalog import Catalog, CatalogEntry, Schema
+from tap_amazon_ads.schema import get_schemas
+
+LOGGER = singer.get_logger()
+
+
+def discover() -> Catalog:
+    """
+    Run the discovery mode, prepare the catalog file and return the catalog.
+    """
+    schemas, field_metadata = get_schemas()
+    catalog = Catalog([])
+
+    for stream_name, schema_dict in schemas.items():
+        try:
+            schema = Schema.from_dict(schema_dict)
+            mdata = field_metadata[stream_name]
+        except Exception as err:
+            LOGGER.error(err)
```

Comment:
Since we are already raising the exception, IMO we don't need to explicitly log it here.